apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessment/assessment.html.ts

```diff
@@ -0,0 +1,167 @@
+
+            <div class="table-responsive">
+              <table id="instanceQuestionGradingTable" class="table table-sm table-hover">
+                <thead>
+                  <tr>
+                    <th>Question</th>
+                    <th>QID</th>
+                    <th>Auto Points</th>
+                    <th>Manual Points</th>
+                    <th colspan="2">Submissions to grade</th>
+                    <th>Grading assigned to</th>
+                    <th>Graded by</th>
+                    <th>Actions</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  ${questions.map((question) => AssessmentQuestionRow({ resLocals, question }))}
+                </tbody>
+              </table>
+            </div>
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function AssessmentQuestionRow({
+  resLocals,
+  question,
+}: {
+  resLocals: Record<string, any>;
+  question: ManualGradingQuestion;
+}) {
+  const showGradingButton =
+    resLocals.authz_data.has_course_instance_permission_edit &&
+    question.num_instance_questions_assigned + question.num_instance_questions_unassigned > 0;
+  const otherAssignedGraders = (question.assigned_graders || [])
+    .filter((u) => !idsEqual(u.user_id, resLocals.authz_data.user.user_id))
+    .map((u) => u.name ?? u.uid);
+  const currentUserName = resLocals.authz_data.user.name ?? resLocals.authz_data.user.uid;
+  const gradingUrl = `${resLocals.urlPrefix}/assessment/${resLocals.assessment.id}/manual_grading/assessment_question/${question.id}`;
+
+  return html`
+    <tr>
+      <td>
+        <a href="${gradingUrl}">
+          ${question.alternative_group_number}.${question.alternative_group_size === 1
+            ? ''
+            : `${question.number_in_alternative_group}.`}
+          ${question.title}
+        </a>
+      </td>
+      <td>${question.qid}</td>
+      <td class="text-center">
+        ${question.max_auto_points
+          ? resLocals.assessment.type === 'Exam'
+            ? (question.points_list || [question.max_manual_points ?? 0])
+                .map((p) => p - (question.max_manual_points ?? 0))
+                .join(',')
+            : (question.init_points ?? 0) - (question.max_manual_points ?? 0)
+          : '—'}
+      </td>
+      <td class="text-center">${question.max_manual_points || '—'}</td>
+      <td class="text-center" data-testid="iq-to-grade-count">
+        ${question.num_instance_questions_to_grade} / ${question.num_instance_questions}
+      </td>
+      <td>
+        ${ProgressBar(question.num_instance_questions_to_grade, question.num_instance_questions)}
+      </td>
+      <td>
+        ${question.num_instance_questions_assigned > 0
+          ? html`
+              <strong class="bg-warning rounded px-1">${currentUserName}</strong
+              >${otherAssignedGraders.length ? ', ' : ''}
+            `
+          : ''}
+        ${otherAssignedGraders.join(', ')}
```

Comment:
Nit: I think this could be cleaner by constructing an array of graders and then joining them. That is, something like this:

```ts
// Ensure that the current user is always listed first.
const assignedGraders = (question.assigned_graders || [])
    .filter((u) => !idsEqual(u.user_id, resLocals.authz_data.user.user_id))
    .map((u) => u.name ?? u.uid);
if (question.num_instance_questions_assigned > 0) {
  assignedGraders.unshift(html`<strong class="bg-warning rounded px-1">${currentUserName}</strong>`);
}

return html`
  ...
  ${assignedGraders.join(', ')}
  ...
`
```

IIRC, there was some trickiness with the whitespace here, which this would also help avoid

Reply:
The problem with that approach is that `.join` would convert the HTML to string and escape the formatting, returning a string.

Reply:
Ahhh, that's a great point. Perhaps we need a `joinHtml` function in `@prairielearn/html` that can do HTML-aware joins? That is:

```ts
function joinHtml(values: HtmlValue[], separator: HtmlValue): HtmlSafetring;
```

If we decide we want this, we can of course save it for a future PR.

FWIW, the code that I based `html` off of has such a function: https://github.com/Janpot/escape-html-template-tag/blob/22b6bf6973a1a7895ae95028726fc323ffdada80/src/index.ts#L18-L26